src/main.py

```diff
@@ -10,36 +10,74 @@
 
 Authors: Zein Youssef and Steven Dropiewski
 """
-import os
-
 from follower import Follower
 from leader import Leader
 from tagrec import TagRecognition
 
+import os
+import subprocess
+import time
+
 
 """
-Determines if a Bluetooth controller is connected. 
+Determines if a Bluetooth controller is connected.
 
 Returns true if a device is connected and false if not.
 """
 def is_controller_connected():
     return os.path.exists('/dev/input/js0')
 
 
+"""
+This function calls the ./bt_toggle script with the specified
+command and address.
+"""
+def bt_toggle(command, address):
+    subprocess.call(['./bt_toggle.sh', command, address])
+
+
 def main():
     tag = TagRecognition()
+    bl = Bluetoothctl()
+    BT_ADDRESS = '5C:BA:37:26:6D:9A'
 
-    # If an ARTag is detected, there is a vehicle ahead of this vehicle so it will follow.
-    # If an ARTag is not detected, no vehicle is present so this vehicle will be a leader.
+    # If an ARTag is detected, the vehicle will disconnect from a Bluetooth
+    # controller. If an ARTag is not detected, the vehicle will connect to
+    # a Bluetooth controller.
     if tag.detect():
-        tag = None
-        follower = Follower()
-        while True:
-            follower.follow()
+        while is_controller_connected():
+            bl.disconnect('5C:BA:37:26:6D:9A')
+        vehicle = Follower()
     else:
-        leader = Leader()
-        while True:
-            leader.lead()
+        while not is_controller_connected():
+            bl.connect('5C:BA:37:26:6D:9A')
+        vehicle = Leader()
+
+    # This loop makes the vehicle move. If the vehicle sees an ARTag then
+    # it is a follower vehicle, otherwise it is a leader vehicle.
+    timer_set = False
+    start_time = time.time()
+    while True:
+        if isinstance(vehicle, Leader):
+            vehicle.lead()
+            if tag.detect():
+                while not is_controller_connected():
+                    bl.disconnect('5C:BA:37:26:6D:9A')
+                vehicle = Follower()
+        else:
+            vehicle.follow()
+            if tag.detect():
```

Comment:
This is what breaks it. Every time we call tag.detect, we get a different result. That's not good. I think what we should do is return the data that tag.detect() in the follower gets, then use that everywhere.